packages/blocks/src/api/matchers.js

```diff
@@ -38,6 +52,6 @@ export function html( selector, multilineTag ) {
 			return value;
 		}
 
-		return match.innerHTML;
+		return replaceInnerHTMLEntities( match.innerHTML );
```

Comment:
I don't think we can change this behaviour at this point. That's why I added a new type. Otherwise we could just return the raw html value here instead?

Reply:
I played with this a bit more today. You're correct we cannot change the behaviour. It even breaks Core block let alone any third party blocks.

I'm now more convinced than ever that having a `raw` source type such as you propose is the best solution.